src/main.c

```diff
@@ -1,49 +1,208 @@
+total_entries--;
 }
-T = eigen_vector(n, sd_matrix);
 
-double* D;
-D = (double*) malloc(n*sizeof(double));
-D = eigen_value(n, sd_matrix);
+int main(){
+    /* Declare a variable to hold the number of entries for a particular time */
+    int n = 0;
+
+    /* Open a file stream */
+    FILE* input_file = fopen("..\\data\\input.csv", "rt");
+    if(input_file == NULL){
+        printf("File could not be opened: %s\n",strerror(errno));
+        return 1;
+    }
+    else{
+    printf("File opened successfully!\n");
+    }
+
+    char line[MAX_LINE_SIZE];
+    sensor_data_t* sensor_description = (sensor_data_t*) malloc(sizeof(sensor_data_t));
+    sensor_data_t sensor_description_list[100];
+
+    /* Ignore the first line containing headers */
+    fgets(line, sizeof(line), input_file);
+
+    /* Parse the input .csv file to store the sensor data in variables */
+    while((fgets(line,sizeof(line),input_file) != NULL)){
+        check(line, &sensor_description_list[total_entries++]);
+    }
+
+    /**
+    * Declare 2 arrays to manage input processing. 
+    * temp_array store the indices which have not been processed yet.
+    * input_array stores the sensor values for a particular time of input.
+    */
+    int temp_array[total_entries];
+    double input_array[total_entries];
+
+    /* Fill the temp_array with indices for input processing. */
+    for(int i = 0; i < total_entries; i++){
+        temp_array[i] = i;
+    }
+
+    /**
+    * Call the functions to compute fused output seperately for different time values.
+    * In each iteration, fused output for one time value is computed.
+    * temp_array is updated in each iteration to store only those indices which have not been processed yet.
+    * The loop continues until all sensor entries are processed.
+    */
+    while(total_entries){
+        for(int j = 1; j < total_entries; j++){
+            int entry = 0;
+            if(sensor_description_list[temp_array[0]].time == sensor_description_list[temp_array[j]].time){
+                input_array[entry] = sensor_description_list[temp_array[j]].value;
+                entry = entry + 1;
+                delete(temp_array,j);
+            }
+            input_array[entry] = sensor_description_list[temp_array[0]].value;
+            float time = sensor_description_list[temp_array[0]].time;
+            entry = entry + 1;
+            delete(temp_array, 0);
+            for(int k = 0; k < entry; k++){
+                printf("\nINPUT ARRAY %f", input_array[k]);
+                n = k + 1;
+            }
+
+            /* Declare a 2-D matrix to store the SDM matrix. */  
+            double** sd_matrix;
+            sd_matrix = (double**) malloc(n*sizeof(double));
+            for(int r = 0; r < n; r++){
+                sd_matrix[r] = (double*) malloc(n*sizeof(double));
+            }
+            sd_matrix = support_degree_matrix(n,input_array);
+
+            /* Declare a 2-D matrix to store eigen vectors. */
+            double** T;
```

Comment:
Variable names should be in lowercase, same follows for line 187.

Reply:
I have named these variables according to the variable names specified in the paper so that it'll be generalized and easy understandable by other group members.